### PR TITLE
Feature/issue 10 add a multi survey application that expects configured subjrect polls on the same account

### PR DESCRIPTION
**Pull request by hodgestar**
Time to review this beast.

### CODE REVIEW

**Comment by hodgestar** [comment]
There is still a small "TODO:" in test_views. I just need to read the new view redirect and figure out what to load / test for.

**Comment by hodgestar** [comment]
The test_views TODO has been done.

File: config/multipoll.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+transport_name: multi_survey_transport
+batch_size: 5
+
+message_store:
+  store_prefix: vumigo.
+riak_manager:
+  bucket_prefix: vumigo.
+vxpolls:
+  prefix: vumigo.
+poll_prefix: vumigo.
```
Comment (smn) [inline config/multipoll.yaml]: why not use `config['vxpolls']['prefix']?`poll_prefix` is just a duplication.

File: go/apps/multi_surveys/tests/test_views.py [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+        # render the form
+        self.assertEqual(len(self.conv_store.list_conversations()), 1)
+        response = self.client.get(reverse('multi_survey:new'))
+        self.assertEqual(response.status_code, 200)
+        # post the form
+        response = self.client.post(reverse('multi_survey:new'), {
+            'subject': 'the subject',
+            'message': 'the message',
+            # 'start_date': datetime.utcnow().strftime('%Y-%m-%d'),
+            # 'start_time': datetime.utcnow().strftime('%H:%M'),
```
Comment (smn) [inline go/apps/multi_surveys/tests/test_views.py]: Can we remove these commented lines if they're not used?

File: go/apps/multi_surveys/tests/test_vumi_app.py [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+    @inlineCallbacks
+    def test_survey_completion(self):
+        yield self.create_contact(u'First', u'Contact',
+            msisdn=u'27831234567', groups=[self.group])
+        self.create_survey(self.conversation)
+        yield self.conversation.start()
+        yield self.complete_survey(self.default_polls)
+
+    @inlineCallbacks
+    def test_surveys_in_succession(self):
```
Comment (smn) [inline go/apps/multi_surveys/tests/test_vumi_app.py:241]: I think this was copied straight over from the tests for `surveys.vumi_app`, does this test anything useful for the multipoll survey app stuff?

File: go/apps/multi_surveys/vumi_app.py [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+from go.vumitools.conversation import ConversationStore
+from vxpolls.multipoll_example import MultiPollApplication
+from vxpolls.manager import PollManager
+from vumi.persist.message_store import MessageStore
+from vumi.persist.txriak_manager import TxRiakManager
+from vumi.message import TransportUserMessage
+from vumi import log
+
+
+class MamaPollApplication(MultiPollApplication):
```
Comment (smn) [inline go/apps/multi_surveys/vumi_app.py:13]: The multi_survey app should not be Mama specific I think? Can we store these attributes in the individual polls instead?

File: config/multipoll.yaml (hunk 1 above)
Comment (smn) [inline config/multipoll.yaml]: Actuall, this isn't used anywhere - perhaps a bit of debris from a merge conflict being resolved?

File: go/apps/multi_surveys/vumi_app.py [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+        if self.control_consumer is not None:
+            yield self.control_consumer.stop()
+            self.control_consumer = None
+
+    def consume_user_message(self, message):
+        helper_metadata = message['helper_metadata']
+        conv_info = helper_metadata.get('conversations')
+        helper_metadata['poll_id'] = 'poll-%s' % (
+            conv_info.get('conversation_key'),)
+        super(MultiSurveyApplication, self).consume_user_message(message)
```
Comment (smn) [inline go/apps/multi_surveys/vumi_app.py:65]: There's very little that subclassing the MultiPollApplication / PollApplication gives us at his point. Should we stop subclassing? That'll prevent us from having to do this copying from dict to dict stuff. I did that with the `surveys.vumi_app` stuff as well.

Comment (smn) [inline go/apps/multi_surveys/vumi_app.py:65]: Actually no I didn't was confused with some other code I think.

File: go/apps/multi_surveys/tests/test_vumi_app.py (hunk 3 above)
Comment (hodgestar) [inline go/apps/multi_surveys/tests/test_vumi_app.py:241]: It was updated after being pulled across -- now it does multiple runs over all the polls (or at least that was the intention). I'm not sure what's supposed to happen if someone comes back.

File: go/apps/multi_surveys/tests/test_views.py (hunk 2 above)
Comment (hodgestar) [inline go/apps/multi_surveys/tests/test_views.py]: Happy for these to be deleted. They probably need to be deleted from the equivalent tests in bulk_message and survey.

File: go/apps/multi_surveys/vumi_app.py (hunk 4 above)
Comment (hodgestar) [inline go/apps/multi_surveys/vumi_app.py:13]: The current poll is somewhat Mama specific, I think. At least the way polls are selected might be Mama specific. Someone will have to read the code or ask David though.

File: go/apps/multi_surveys/tests/test_vumi_app.py (hunk 3 above)
Comment (smn) [inline go/apps/multi_surveys/tests/test_vumi_app.py:241]: Ah cool, missed that.

File: go/apps/multi_surveys/tests/test_views.py (hunk 2 above)
Comment (smn) [inline go/apps/multi_surveys/tests/test_views.py]: Done

File: go/apps/multi_surveys/vumi_app.py (hunk 4 above)
Comment (smn) [inline go/apps/multi_surveys/vumi_app.py:13]: Leaving this for now, there's a way to store this information on the poll though. Should look at that post-monday's event.

**Pull request merged**